lib/orderbook/TradingPair.ts

```diff
@@ -317,27 +323,36 @@ class TradingPair {
           const matchedMakerOrder = TradingPair.splitOrderByQuantity(makerOrder, matchingQuantity);
           this.logger.debug(`reduced order ${makerOrder.id} by ${matchingQuantity} quantity while matching order ${takerOrder.id}`);
           matches.push({ maker: matchedMakerOrder, taker: remainingOrder });
-        } else if (makerFullyMatched) {
+        } else if (makerFullyMatched || makerAvailableQuantityFullyMatched) {
```

Comment:
I think this can just check `makerAvailableQuantityFullyMatched` because I believe it will be true any time `makerFullyMatched` is true.